corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -266,6 +275,98 @@ public int dropTable(String namespace, String tablename) {
+            runtime.getTableRegistry().getTableNameToKeyValueTypeTable().get(tableNameProto);
+        Any defaultKeyAny = keyValueDefaultRecord.getKey();
+        Any defaultValueAny = keyValueDefaultRecord.getValue();
+
+        DynamicMessage keyMsg = createMsgFromJson(defaultKeyAny, keyToEdit);
+        DynamicMessage newValueMsg = createMsgFromJson(defaultValueAny,
+            newRecord);
+
+        if (keyMsg == null || newValueMsg == null) {
+            return 0;
+        }
+
+        for (CorfuDynamicKey key : table.keySet()) {
+            if (Objects.equals(keyMsg.toString(), key.getKey().toString())) {
+                log.info("Editing record with Key {} in table {} and " +
+                        "namespace {} with new record {}.  Stream Id {}.",
+                    keyToEdit, tableName, namespace, newRecord, streamUUID);
+                String payloadTypeUrl = table.get(key).getPayloadTypeUrl();
+                String metadataTypeUrl = table.get(key).getMetadataTypeUrl();
+                DynamicMessage metadata = table.get(key).getMetadata();
+                table.put(key, new CorfuDynamicRecord(payloadTypeUrl,
+                    newValueMsg, metadataTypeUrl, metadata));
+                log.info("\n======================\n");
+                return 1;
+            }
+        }
+        log.info("Record with key {} not found in table {} and namespace {}. " +
+            " Stream Id {}.", keyToEdit, tableName, namespace, streamUUID);
+        log.info("\n======================\n");
+        return 0;
+    }
+
+    private DynamicMessage createMsgFromJson(Any anyMsg, String jsonString) {
+        Descriptors.FileDescriptor fileDescriptor;
+        try {
+            ConcurrentMap<String, String> messagesFdProtoNameMap =
+                dynamicProtobufSerializer.getMessagesFdProtoNameMap();
+            String fullMessageName =
+                dynamicProtobufSerializer.getFullMessageName(anyMsg);
+
+            fileDescriptor = dynamicProtobufSerializer.getDescriptor(
+                messagesFdProtoNameMap.get(fullMessageName));
+        } catch (Descriptors.DescriptorValidationException e) {
+            log.warn("DescriptorValidationException thrown {}", e);
+            return null;
+        }
+        Descriptors.Descriptor descriptor =
+            fileDescriptor.findMessageTypeByName(
+                dynamicProtobufSerializer.getMessageName(anyMsg));
+        DynamicMessage.Builder builder;
+        try {
+            builder = DynamicMessage.parseFrom(descriptor,
+                anyMsg.getValue()).toBuilder();
+        } catch (InvalidProtocolBufferException e) {
+            log.warn("Unable to Parse Key {}", anyMsg.getValue());
+            return null;
+        }
+
+        try {
+            JsonFormat.parser().merge(jsonString, builder);
+        } catch(InvalidProtocolBufferException e) {
+            log.warn("Unable to Parse String {}", jsonString);
+            return null;
```

Comment:
This logic seems generic enough to be pushed to the serializer.

Reply:
DynamicProtobufSerializer is for conversion to/from bytebuf to/from dynamic messages.  Json string is very specific to the editor use case.  Should we still move this there?  Let me know your thoughts.

Reply:
moved to DynamicProtobufSerializer